packages/ngx-view3d/projects/ngx-view3d/src/lib/ngx-view3d.component.ts

```diff
@@ -0,0 +1,158 @@
+  AREndEvent,
+  ARModelPlacedEvent,
+  QuickLookTapEvent,
+  View3DOptions
+} from "@egjs/view3d";
+import { optionNames, optionInputs, setterNames } from "./const";
+
+@Component({
+  selector: "ngx-view3d, [NgxView3D]",
+  template: `
+    <canvas #canvas [ngClass]="_canvasElClass"></canvas>
+    <ng-content></ng-content>
+  `,
+  host: {
+    style: "display: block;",
+    class: DEFAULT_CLASS.WRAPPER
+  },
+  inputs: optionInputs
+})
+export class NgxView3DComponent
+  implements AfterViewInit, OnDestroy, OnChanges {
+  @Input() public canvasClass: string;
+
+  @Output("ready") public readyEmitter = new EventEmitter<ReadyEvent>();
+  @Output("loadStart") public loadStartEmitter = new EventEmitter<LoadStartEvent>();
+  @Output("load") public loadEmitter = new EventEmitter<LoadEvent>();
+  @Output("loadError") public loadErrorEmitter = new EventEmitter<LoadErrorEvent>();
+  @Output("loadFinish") public loadFinishEmitter = new EventEmitter<LoadFinishEvent>();
+  @Output("modelChange") public modelChangeEmitter = new EventEmitter<ModelChangeEvent>();
+  @Output("resize") public resizeEmitter = new EventEmitter<ResizeEvent>();
+  @Output("beforeRender") public beforeRenderEmitter = new EventEmitter<BeforeRenderEvent>();
+  @Output("render") public renderEmitter = new EventEmitter<RenderEvent>();
+  @Output("progress") public progressEmitter = new EventEmitter<ProgressEvent>();
+  @Output("inputStart") public inputStartEmitter = new EventEmitter<InputStartEvent>();
+  @Output("inputEnd") public inputEndEmitter = new EventEmitter<InputEndEvent>();
+  @Output("cameraChange") public cameraChangeEmitter = new EventEmitter<CameraChangeEvent>();
+  @Output("animationStart") public animationStartEmitter = new EventEmitter<AnimationStartEvent>();
+  @Output("animationLoop") public animationLoopEmitter = new EventEmitter<AnimationLoopEvent>();
+  @Output("animationFinished") public animationFinishedEmitter = new EventEmitter<AnimationFinishedEvent>();
+  @Output("annotationFocus") public annotationFocusEmitter = new EventEmitter<AnnotationFocusEvent>();
+  @Output("annotationUnfocus") public annotationUnfocusEmitter = new EventEmitter<AnnotationUnfocusEvent>();
+  @Output("arStart") public arStartEmitter = new EventEmitter<ARStartEvent>();
+  @Output("arEnd") public arEndEmitter = new EventEmitter<AREndEvent>();
+  @Output("arModelPlaced") public arModelPlacedEmitter = new EventEmitter<ARModelPlacedEvent>();
+  @Output("quickLookTap") public quickLookTapEmitter = new EventEmitter<QuickLookTapEvent>();
+
+  @ViewChild("canvas") public canvas: ElementRef<HTMLCanvasElement>;
+
+  public get element() { return this._elRef.nativeElement; }
+  private get _canvasElClass() { return `${DEFAULT_CLASS.CANVAS} ${this.canvasClass ?? ""}`.trim(); }
+  private _view3D: VanillaView3D;
+
+  public constructor(
+    private _elRef: ElementRef<HTMLElement>,
+    @Inject(PLATFORM_ID) private _platformId
+  ) {
+    this._view3D = null;
+
+    withMethods(this, "_view3D");
```

Comment:
There is no type for the method.

Reply:
There is a bug in the Angular compiler, so interface (with method type) => component is required.

Reply:
What do you mean?

Reply:
```ts
class ConveyerInterface {}
interface ConveyerInterface extends ConveyerMethods {}

@Component({})
class ConveyerComponent extends ConveyerInterface {}
```

Reply:
Now I get it.
Good point, I guess I omitted that while testing something.